### PR TITLE
Add multiopt class for evalset

### PR DESCRIPTION
**Pull request by kevintee**
`MultioptTestFunction` class for testing. More functions will be added in the future, but wanted to get initial thoughts (particularly if there's anything missing in the `__init__` function).

### CODE REVIEW

File: evalset/multiopt_test_funcs.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
+from evalset.test_funcs import TestFunction, lzip
+import numpy
+
+class MultioptTestFunction(TestFunction):
+
+  def __init__(self, dim):
+    super(MultioptTestFunction, self).__init__(dim)
+    self.local_minima = [] # Sorted in increasing order of function value at the local minima
```
Comment (pfhayes) [inline evalset/multiopt_test_funcs.py]: couldn't this be enforced automatically by calling do_evaluate on all the minima?

File: evalset/multiopt_test_funcs.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+class LowDMixtureOfGaussians(MultioptTestFunction):
+
+  def __init__(self, dim=2):
+    assert dim == 2
+    super(KevinTest, self).__init__(dim)
```
Comment (pfhayes) [inline evalset/multiopt_test_funcs.py]: KevinTest

File: evalset/multiopt_test_funcs.py (hunk 1 above)
Comment (kevintee) [reply evalset/multiopt_test_funcs.py]: I should rename this as `local_minima_loc` (analogous to `min_loc`) in the original `evalset`.

**Comment by kevintee** [comment]
cc @mikemccourt

File: evalset/multiopt_test_funcs.py [hunk 3]
```diff
@@ -0,0 +1,25 @@
+from evalset.test_funcs import TestFunction, lzip
+import numpy
+
+class MultioptTestFunction(TestFunction):
+
+  def __init__(self, dim):
+    super(MultioptTestFunction, self).__init__(dim)
+    self.local_minima_loc = [] # Sorted in increasing order of function value at the local minima
+
+class LowDMixtureOfGaussians(MultioptTestFunction):
```
Comment (mikemccourt) [inline evalset/multiopt_test_funcs.py]: Are you going to be adding more of these?  If so, you might want to append 01 to this.  Or maybe make this an extensible class.  If it'll be a lot of different functions then it's all good.

Comment (kevintee) [reply evalset/multiopt_test_funcs.py]: Both good ideas. What's the naming convention for functions?

**Comment by mikemccourt** [comment]
lgtm

**Pull request merged**